tenable/io/v3/vm/tags/api.py

```diff
@@ -0,0 +1,746 @@
+                combined together.  ``and`` will inform the API that all of
+                the filter conditions must be met whereas ``or`` would mean
+                that if any of the conditions are met. Default is ``and``
+            all_users_permissions (list, optional):
+                List of the minimum set of permissions all users have on the
+                current tag.
+                Possible values are ALL, CAN_EDIT, and CAN_SET_PERMISSIONS.
+            current_domain_permissions (list, optional):
+                List of user and group-specific permissions for the current tag
+                current_domain_permissions are list of tuples in the form of
+                ('ID', 'NAME', 'TYPE', 'PERMISSIONS')
+                the TYPE can be either `USER` or `GROUP` and the
+                PERMISSIONS can be `ALL`, `CAN_EDIT` or `CAN_SET_PERMISSIONS`
+                any one or all in list
+
+                Examples::
+                    - ``(uuid, 'user@company.com', 'USER', ['CAN_EDIT'])``
+
+        Returns:
+            :obj:`dict`:
+                Tag value resource record.
+
+        Examples:
+            >>> tio.v3.vm.tags.edit('00000000-0000-0000-0000-000000000000',
+            ...     value='NewValueName')
+        '''
+
+        payload: dict = {
+            'value': value,
+            'description': description
+        }
+
+        # get existing values of tag
+        current = self.details(value_id=value_id)
+        current_access_control = current['access_control']
+
+        # created copy of current access control which will be used
+        # to compare any changes done in permissions
+        access_control = current_access_control.copy()
+
+        # initialize access controls
+        payload['access_control'] = dict()
+
+        # Set all users permission
+        if all_users_permissions is not None:
+            current_access_control['all_users_permissions'] = \
+                all_users_permissions
+
+        # run current_domain_permissions through permission parser
+        if current_domain_permissions is not None:
+            current_access_control['current_domain_permissions'] = \
+                current_domain_permissions
+
+        # update payload access control with new values
+        payload['access_control'] = dict_merge(
+            payload['access_control'],
+            current_access_control
+        )
+
+        # We need to pick current value of version if available or set default
+        # value to 0.
+        # this value will be incremented when permissions are updated
+        if 'version' in current['access_control']:
+            current_version = current['access_control']['version']
+        else:
+            current_version = 0
+
+        # version value must be incremented each time the
+        # permissions are updated
+        if payload['access_control'] != access_control:
+            payload['access_control']['version'] = current_version + 1
```

Comment:
Doesn't the concerned endpoint in the backend support partial updates and updates versions ? Calculating version in SDK seems to be heavyweight lifting and is not ideal in concurrent env, as it'll result in version conflicts during concurrent updates.